From 653009d72acf16cdd2203824199b91bfd8baee47 Mon Sep 17 00:00:00 2001
From: Shanavas M <shanavas.m2@gmail.com>
Date: Fri, 28 Apr 2017 14:20:41 +0300
Subject: [PATCH] Get tarball root directory from member filename

Some packages like 'editorconfig' has tarball root directory name
other than 'package'

Fixes #61
---
 npm2deb/__init__.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- a/npm2deb/__init__.py
+++ b/npm2deb/__init__.py
@@ -434,11 +434,18 @@
 
         tarball_file = info[1].strip('\n')
         tarball = tarfile.open(tarball_file)
+        # get the root directory name
+        root_dir = tarball.getnames()[0]
+        # extract root directory name if memberfile contains '/'
+        index_of_slash = root_dir.find('/')
+        if index_of_slash != -1:
+            root_dir = root_dir[:index_of_slash]
+
         tarball.extractall()
         tarball.close()
 
         # rename extracted directory
-        _os.rename('package', self.name)
+        _os.rename(root_dir, self.name)
         # remove tarball file
         _os.remove(tarball_file)
 
